runtime/jcl/common/jclcinit.c

```diff
@@ -158,35 +133,34 @@ jint computeFullVersionString(J9JavaVM* vm)
 			aotEnabled = 1;
 		}
 	}
-	strcat(fullversion, " (JIT ");
-	strcat(fullversion, jitEnabled ? "en" : "dis");
-	strcat(vminfo, " (JIT ");
-	strcat(vminfo, jitEnabled ? "en" : "dis");
-	strcat(fullversion, "abled, AOT ");
-	strcat(fullversion, aotEnabled ? "en" : "dis");
-	strcat(vminfo, "abled, AOT ");
-	strcat(vminfo, aotEnabled ? "en" : "dis");
-	strcat(fullversion, "abled)\nOpenJ9   - ");
-	strcat(vminfo, "abled)\nOpenJ9   - ");
+	#define JIT_INFO " (JIT %sabled, AOT %sabled)\nOpenJ9   - "
 #endif /* J9VM_INTERP_NATIVE_SUPPORT */
 
-	vmVersion = J9VM_VERSION_STRING;
-	strcat(fullversion, vmVersion);
-	strcat(vminfo, vmVersion);
-
 #if defined(J9VM_GC_MODRON_GC)
-	gcVersion = OMR_VERSION_STRING;
-	strcat(fullversion, "\nOMR      - ");
-	strcat(fullversion, gcVersion);
-	strcat(vminfo, "\nOMR      - ");
-	strcat(vminfo, gcVersion);
+	#define OMR_INFO "\nOMR      - " OMR_VERSION_STRING
+#else
+	#define OMR_INFO ""
 #endif /* J9VM_GC_MODRON_GC */
 
 #if defined(VENDOR_SHORT_NAME) && defined(VENDOR_SHA)
-	strcat(fullversion, "\n" VENDOR_SHORT_NAME "      - " VENDOR_SHA);
-	strcat(vminfo, "\n" VENDOR_SHORT_NAME "      - " VENDOR_SHA);
+	#define VENDOR_INFO "\n" VENDOR_SHORT_NAME "      - " VENDOR_SHA);
+#else
+	#define VENDOR_INFO ""
 #endif /* VENDOR_SHORT_NAME && VENDOR_SHA */
 
+#if defined(J9VM_INTERP_NATIVE_SUPPORT)
+	j9str_printf(PORTLIB, fullversion, sizeof(fullversion), "JRE %s IBM J9 %s %s %s " MEM_INFO "%s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO, j2se_version_info, EsVersionString, (NULL != osname ? osname : " "), osarch, EsBuildVersionString, (jitEnabled ? "en" : "dis"), (aotEnabled ? "en" : "dis"));
+	j9str_printf(PORTLIB, vminfo, sizeof(vminfo), "JRE %s %s %s" MEM_INFO "%s" JIT_INFO J9VM_VERSION_STRING OMR_INFO VENDOR_INFO, j2se_version_info, (NULL != osname ? osname : " "), osarch, EsBuildVersionString, (jitEnabled ? "en" : "dis"), (aotEnabled ? "en" : "dis"));
+#undef JIT_INFO
+#else
+	j9str_printf(PORTLIB, fullversion, sizeof(fullversion), "JRE %s IBM J9 %s %s %s " MEM_INFO "%s" J9VM_VERSION_STRING OMR_INFO VENDOR_INFO, j2se_version_info, EsVersionString, (NULL != osname ? osname : " "), osarch, EsBuildVersionString);
+	j9str_printf(PORTLIB, vminfo, sizeof(vminfo), "JRE %s %s %s" MEM_INFO "%s" J9VM_VERSION_STRING OMR_INFO VENDOR_INFO, j2se_version_info, (NULL != osname ? osname : " "), osarch, EsBuildVersionString);
```

Comment:
The lines calling `j9str_printf` are too long, please wrap at a reasonable width.